src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -591,7 +410,12 @@ BATT_SMBUS::cycle()
 
 		// publish to orb
 		if (_batt_topic != nullptr) {
-			orb_publish(_batt_orb_id, _batt_topic, &new_report);
+			if (success == true) {
+				orb_publish(_batt_orb_id, _batt_topic, &new_report);
```

Comment:
I think in this case, the fuel gauge provides such a better measurement of voltage, current, and remaining capacity that I'd rather have the ADC measurement completely disabled to avoid any confusion on the companion computer.

Reply:
Yes and that's what the battery source parameter can set. Once they actually get to the point of publishing using orb_publish_auto ensures we don't have multiple publishers on the same topic instance.

Later BAT_SOURCE or another parameter could be expanded to properly support having both power modules and smart batteries connected and used simultaneously.